ledger-service/http-json/src/main/scala/com/digitalasset/http/HttpService.scala

```diff
@@ -134,12 +140,20 @@ object HttpService extends StrictLogging {
     bindingEt.run: Future[Error \/ ServerBinding]
   }
 
-  private[http] def loadPackageStoreUpdates(packageClient: PackageClient, token: Option[String])(
+  private[http] def loadPackageStoreUpdates(
+      packageClient: PackageClient,
+      tokenHolder: Option[TokenHolder])(
       implicit ec: ExecutionContext): PackageService.ReloadPackageStore =
-    (ids: Set[String]) =>
+    (ids: Set[String]) => {
+      val token =
+        tokenHolder.flatMap { holder =>
+          holder.refresh()
```

Comment:
@stefanobaghino-da you are not checking for the return value here. `None` means failure as I understand. This should be returned as `PackageService.Error`